### DISCUSSION TITLE
Use collections as user quota

### DISCUSSION THREAD

**Discussion by jvinolas** [seaweedfs/seaweedfs#2022] [Ideas]
Is it possible to use collections (https://github.com/chrislusf/seaweedfs/wiki/Optimization#collection-as-a-simple-name-space) as a kind of quota for users if we use username as collection name?
In that document it states that "Each collection will have its dedicated volumes, and they will not share the same volume." so I was thinking if we could limit the maximum volumes 'V' of size 'S' as per collection and then each user will have a quota of V*S.

If that is possible, how to achieve it?

**Comment by chrislusf** [comment]
Currently there are no quota controls implemented.

I am planning to add an enterprise/paid add-on component to manage quota. Let me know if you are interested.

**Reply by ningfdx** [reply]
Hello chirslusf,
I'm intresting at the feature of limit on a filer directory. Is there any update about it? I walk through the issues and git commit, didn't found any plan or feature about that (only collection quota was supported..)

I think it is posible to achieve in the filer since the directory listing is not that expencive op, and depands on the good design of Pub/Sub of filer's client to filer, it will be efficient and reliable.

Well, if "enterprise/paid add-on component to manage quota" is done, I'm glad to pay for that component or plugin, I look forward to your reply, HAHA

**Reply by chrislusf** [reply]
https://github.com/chrislusf/seaweedfs/wiki/S3-Bucket-Quota
